sim-lib/src/lib.rs

```diff
@@ -958,18 +964,44 @@ impl PaymentResultLogger {
         }
 
         self.total_sent += details.amount_msat;
-        self.call_count += 1;
+    }
+}
 
-        if self.call_count % self.log_interval == 0 || self.call_count == 0 {
-            let total_payments = self.success_payment + self.failed_payment;
-            log::info!(
-                "Processed {} payments sending {} msat total with {}% success rate.",
-                total_payments,
-                self.total_sent,
-                (self.success_payment * 100 / total_payments)
-            );
+impl Display for PaymentResultLogger {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let total_payments = self.success_payment + self.failed_payment;
+        write!(
+            f,
+            "Processed {} payments sending {} msat total with {:.2}% success rate.",
+            total_payments,
+            self.total_sent,
+            (self.success_payment as f64 / total_payments as f64) * 100.0
+        )
+    }
+}
```

Comment:
It could now be the case that we sample before a single payment has been tracked yet, meaning that we'd get `0/0` when reporting the success rate, yielding `NaN`:

```
INFO  [sim_lib] Processed 0 payments sending 0 msat total with NaN% success rate.
```

We should treat that as a special case:

```suggestion
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        let total_payments = self.success_payment + self.failed_payment;
        let rate = self.success_payment as f64 / total_payments as f64;
        if rate.is_nan() {
            write!(
                f,
                "Processed {} payments sending {} msat total.",
                total_payments, self.total_sent
            )
        } else {
            write!(
                f,
                "Processed {} payments sending {} msat total with {:.2}% success rate.",
                total_payments,
                self.total_sent,
                rate * 100.0
            )
        }
    }
```

Reply:
Rate is only NaN if we haven't sent anything yet, so the special case will just print "Processed 0 payments sending 0 msat total." IMO it's not worth the effort to convey essentially the same information in double the LOC, I think it's reasonably understandable as is.